x-pack/filebeat/input/httpjson/date_cursor.go

```diff
@@ -23,10 +25,10 @@ type dateCursor struct {
 	dateFormat      string
 
 	value    string
-	valueTpl *Template
+	valueTpl *template.Template
```

Comment:
this type mixes the computation/extraction of the cursor and the actual data. In case we want to store (or pass around) the data, consider to split the actual data (in the case 'value'), such that the 'computation' becomes immutable.